src/ypd/servlets/user_servlet.py

```diff
@@ -1,27 +1,93 @@
-from flask import flash, redirect, render_template, request, url_for
-from flask_classy import FlaskView, route
-from flask_login import login_required, login_user, logout_user
+import os
+from datetime import datetime, timedelta
+
+from flask import (Markup, current_app, flash, redirect, render_template, request, url_for)
+from flask_login import current_user, login_required, login_user, logout_user
 from sqlalchemy.exc import IntegrityError
 
-from ypd.form.user_form import LoginForm, RegistrationForm
+from flask_classy import FlaskView, route
+from flask_mail import Mail, Message
+from ypd.form.user_form import (ChangePasswordForm, LoginForm, RegistrationForm, ValidateUserForm)
 from ypd.model.user import User, UserType
 
+# from ..server import mail
+
 """A class that represents User creation routes"""
 class UserView(FlaskView):
+    msg = ""
     # Routes work
     @route('/login/', methods=['POST', 'GET'])
     def login(self):
         form = LoginForm()
-        if request.method == 'POST' and form.validate_on_submit:
+        if form.validate_on_submit:
             try:
-                user = User.log_in(username=form.username.data, password=form.password.data)
-                login_user(user, remember=form.remember.data)
+                user = User.log_in(form.username.data, form.password.data)
+                login_user(user, remember=form.remember.data, duration=timedelta(minutes=30.0))
                 return redirect(url_for('IndexView:get'))
             except ValueError as e:
                 flash(str(e))
 
         return render_template('login.html', form=form)
 
+    @route('/change/', methods=['POST', 'GET'])
+    @login_required
+    def changePassword(self):
+        form = ChangePasswordForm()
+        if form.validate_on_submit:
+            try:
+                # print(os.environ.get('MAIL_USERNAME'))
+                # print(os.environ.get('MAIL_PASSWORD'))
+                # current_user is User.log_in(current_user.username, form.old_password.data)
+                if current_user.get_email():
+                    current_user.update_password(form.new_password.data, form.confirm_new.data)
+
+                    '''Recommended to create environment variables for mail_username and mail_password for security/convenience reasons'''                    
+                    current_app.config.update({"MAIL_SERVER": 'smtp.gmail.com', 'MAIL_PORT': 587, 'MAIL_USERNAME': os.environ.get('MAIL_USERNAME'), 'MAIL_PASSWORD': os.environ.get('MAIL_PASSWORD'), 'MAIL_DEFAULT_SENDER': os.environ.get('MAIL_USERNAME'), 'MAIL_USE_TLS' : True, 'MAIL_USE_SSL': False})
+                    mail = Mail()
```

Comment:
We should probably use YPD specific environment variables. E.g. YPD_MAIL_USERNAME. In order to avoid conflicts with other environment variables that may be set on the server.

Reply:
Tis done